rest/src/main/kotlin/builder/component/SelectMenuBuilder.kt

```diff
@@ -69,3 +73,45 @@ public class SelectMenuBuilder(
     }
 
 }
+
+@KordDsl
+public class StringSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.StringSelect
+}
+
+@KordDsl
+public class UserSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.UserSelect
+}
+
+@KordDsl
+public class RoleSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.RoleSelect
+}
+
+@KordDsl
+public class MentionableSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.MentionableSelect
+}
+
+@KordDsl
+public class ChannelSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.ChannelSelect
+
+    private var _channelTypes: Optional<List<ChannelType>> = Optional.Missing()
+
+    public var channelTypes: List<ChannelType>? by ::_channelTypes.delegate()
+
+    override fun build(): DiscordChatComponent {
+        return DiscordChatComponent(
+            type,
+            customId = Optional(customId),
+            disabled = _disabled,
+            placeholder = _placeholder,
+            minValues = OptionalInt.Value(allowedValues.start),
+            maxValues = OptionalInt.Value(allowedValues.endInclusive),
+            options = Optional(options.map { it.build() }),
+            channelTypes = _channelTypes
+        )
+    }
```

Comment:
I would instead add the following in the supertype instead
```kotlin
protected open fun buildChannelTypes(): Optional<List<ChannelType>> = Optional.Missing()
```
that is then used in the `build()` of the supertypes and override it here
```kotlin
override fun buildChannelTypes() = _channelTypes
```
This way we don't have to duplicate the whole logic of `build()` here.

Reply:
Take a look at the [automod builders](https://github.com/kordlib/kord/blob/828c924eccf89fe904593250ea216c0c8019a227/rest/src/main/kotlin/builder/automoderation/AutoModerationRuleCreateBuilder.kt#L31), they already use this pattern.